daffodil-udf/src/test/scala/org/sgoodudfs/example/SupportedTypesFunctions/SupportedTypesFunctionsProvider.scala

```diff
@@ -0,0 +1,236 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sgoodudfs.example.SupportedTypesFunctions
+
+import org.apache.daffodil.udf.UserDefinedFunction
+import org.apache.daffodil.udf.UserDefinedFunctionProvider
+import org.apache.daffodil.udf.UserDefinedFunctionIdentification
+
+/**
+ * Example User Defined Function Provider in Scala
```

Comment:
Can you explain what this file is about?  It's weird and immediately caught my eye. This comment is boilerplate. You are testing all the implicit boxing conversions inbound and for return types right? A user would never do anything like this. Right?

Reply:
Ah so, this file is supposed to be contain UDFs that exercise the types we support. So checking that we can pass in (params) and receive back the types we claim to support.